src/coreclr/jit/morph.cpp

```diff
@@ -11909,6 +11909,22 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
             return fgMorphCast(tree);
 
         case GT_MUL:
+            // -a * C => a * -C, where C is constant
+            // MUL(NEG(a), C) => MUL(a, NEG(C))
+            if (fgGlobalMorph && op1->OperIs(GT_NEG) && !op1->gtGetOp1()->IsCnsIntOrI() && op2->IsCnsIntOrI() &&
+                !op2->IsIconHandle() && op2->AsIntCon()->IconValue() != 1 && op2->AsIntCon()->IconValue() != 0 &&
```

Comment:
Why do you check `fgGlobalMorph` here? Did you want to use `!gtIsActiveCSE_Candidate` ? `gtIsActiveCSE_Candidate` means it is not CSE phase (it could be a global morph phase instead) or this tree is not the current candidate that could be dangerous to remove.

Reply:
I checked `fgGlobalMorph` because it was checked in #43921, which is what I based most of this work on. I can use `!gtIsActiveCSE_Candidate` if that would be better.

Still learning about the CoreCLR JIT, I greatly appreciate the advice :).

Reply:
On second thought I think `fgGlobalMorph` check is fine here because you are changing node value without updating its VN.
An alternative could be to do the optimization with `!gtIsActiveCSE_Candidate` but create a new tree for the constant:
```
tree->AsOp()->gtOp2 = gtNewIconNode(-op2->AsIntCon()->IconValue(), op2->TypeGet());
DEBUG_DESTROY_NODE(op2);
```
it would allow the optimization to happen in other phases, so I think it would be better.